storm-server/src/test/java/org/apache/storm/TickTupleTest.java

```diff
@@ -18,85 +18,103 @@
-            cluster.advanceClusterTime(2);
-            Assert.assertTrue("Test is passed", true);
-        } finally {
-            cluster.close();
+            try (ILocalTopology topo = cluster.submitTopology("test", topoConf,  topology)) {
+                //Give the cluster some time to come up
+                cluster.advanceClusterTime(40);
+                tickTupleTimes.clear();
+                cluster.advanceClusterTime(1);
+                assertEquals(41000, tickTupleTimes.poll(100, TimeUnit.MILLISECONDS).longValue());
+                cluster.advanceClusterTime(1);
+                assertEquals(42000, tickTupleTimes.poll(100, TimeUnit.MILLISECONDS).longValue());
+                cluster.advanceClusterTime(1);
+                assertEquals(43000, tickTupleTimes.poll(100, TimeUnit.MILLISECONDS).longValue());
+                cluster.advanceClusterTime(1);
+                assertEquals(44000, tickTupleTimes.poll(100, TimeUnit.MILLISECONDS).longValue());
+                cluster.advanceClusterTime(1);
+                assertEquals(45000, tickTupleTimes.poll(100, TimeUnit.MILLISECONDS).longValue());
+            }
+            assertNull("The bolt got a tuple that is not a tick tuple " + nonTickTuple.get(), nonTickTuple.get());
         }
-
     }
 
-    private IRichSpout makeNoOpSpout() {
-        return new BaseRichSpout() {
-            @Override
-            public void declareOutputFields(OutputFieldsDeclarer declarer) {
-                declarer.declare(new Fields("tuple"));
-            }
-
-            @Override
-            public void open(Map<String, Object> conf, TopologyContext context, SpoutOutputCollector collector) {
-            }
+    private static class NoopSpout extends BaseRichSpout {
+        @Override
+        public void declareOutputFields(OutputFieldsDeclarer declarer) {
+            declarer.declare(new Fields("tuple"));
+        }
 
-            @Override
-            public void nextTuple() {
-            }
+        @Override
+        public void open(Map<String, Object> conf, TopologyContext context, SpoutOutputCollector collector) {
+        }
 
-            private void writeObject(java.io.ObjectOutputStream stream) {
-            }
-        };
+        @Override
+        public void nextTuple() {
+        }
     }
 
-    private BaseRichBolt makeNoOpBolt() {
-        return new BaseRichBolt() {
-            @Override
-            public void prepare(Map<String, Object> conf, TopologyContext topologyContext, OutputCollector outputCollector) {}
-            @Override
-            public void execute(Tuple tuple) {}
+    private static class NoopBlot extends BaseRichBolt {
+        @Override
+        public void prepare(Map<String, Object> conf, TopologyContext topologyContext, OutputCollector outputCollector) {}
 
-            @Override
-            public void cleanup() { }
+        @Override
+        public void execute(Tuple tuple) {
+            LOG.error("GOT {}", tuple);
```

Comment:
Nit: Info or debug might be nicer so a grep for errors in the logs doesn't produce this line.